source/index.ts

```diff
@@ -184,6 +190,28 @@ class APIClient {
         return response.access_token;
     }
 
+    async _waitUploadSuccess(response: ItemResource, maxAwaitInProgressResponseSeconds: number): Promise<ItemResource> {
+        if (response.uploadState !== 'IN_PROGRESS') {
+            // We can return the response immediately if the upload is not in progress
+            return response;
+        }
+
+        const maxWait = maxAwaitInProgressResponseSeconds * 1000;
+        const retryInterval = 2000;
+
+        if (maxWait < 0) {
+            throw new Error('Upload is still in progress after maximum retries');
```

Comment:
I think the retry logic should not affect the result. If the response is `IN_PROGRESS`, then it should return `IN_PROGRESS`.

Basically the first condition in this should look something like:

- if is not in progress, or
- if don't have time for another retry

then `return response` and let the program continue.